File: bench/prime.go
package main

import (
    "fmt"
)

func IsPrime(n int) bool {
    for j := 2; j * j < n + 1; j++ {
        if n % j == 0 {
            return false
        }
    }
    return true
}

func main() {
    for i := 2; i < 100000; i++ {
        if IsPrime(i) {
            fmt.Printf("%d\n", i)
        }
    }
}

File: bench/prime.php
<?php
$maxnum = 100000;
$i = 2;

function is_prime($n) {
    $j = 2;
    while ($j * $j < $n + 1) {
        if ($n % $j == 0) {
            return false;
        }
        $j += 1;
    }
    return true;
}

while ($i <= $maxnum) {
    if (is_prime($i)) {
        printf("%d\n", $i);
    }
    $i += 1;
}

